// ngx-filesaver/esm5/src/filesaver.module.js
import * as ɵngcc0 from '@angular/core';
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule } from '@angular/core';
import { FileSaverDirective } from './filesaver.directive';
var FileSaverModule = /** @class */ (function () {
    function FileSaverModule() {
    }
FileSaverModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FileSaverModule });
FileSaverModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FileSaverModule_Factory(t) { return new (t || FileSaverModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FileSaverModule, { declarations: function () { return [FileSaverDirective]; }, exports: function () { return [FileSaverDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FileSaverModule, [{
        type: NgModule,
        args: [{
                declarations: [FileSaverDirective],
                exports: [FileSaverDirective]
            }]
    }], function () { return []; }, null); })();
    return FileSaverModule;
}());
export { FileSaverModule };
