// angular-responsive-carousel/esm2015/lib/carousel.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CarouselComponent } from './carousel.component';
import * as ɵngcc0 from '@angular/core';
export class IvyCarouselModule {
}
IvyCarouselModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IvyCarouselModule });
IvyCarouselModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IvyCarouselModule_Factory(t) { return new (t || IvyCarouselModule)(); }, providers: [], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IvyCarouselModule, { declarations: function () { return [CarouselComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CarouselComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(IvyCarouselModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    CarouselComponent
                ],
                imports: [
                    CommonModule
                ],
                exports: [
                    CarouselComponent
                ],
                providers: [],
                bootstrap: [],
                entryComponents: [
                    CarouselComponent
                ]
            }]
    }], null, null); })();
